plugins/rocksdb/src/main/java/tech/pegasys/pantheon/plugin/services/storage/rocksdb/RocksDBPlugin.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.plugin.services.storage.rocksdb;
+
+import tech.pegasys.pantheon.metrics.MetricsSystem;
+import tech.pegasys.pantheon.metrics.noop.NoOpMetricsSystem;
+import tech.pegasys.pantheon.plugin.PantheonContext;
+import tech.pegasys.pantheon.plugin.PantheonPlugin;
+import tech.pegasys.pantheon.plugin.services.PantheonConfiguration;
+import tech.pegasys.pantheon.plugin.services.PicoCLIOptions;
+import tech.pegasys.pantheon.plugin.services.StorageService;
+import tech.pegasys.pantheon.plugin.services.storage.SegmentIdentifier;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+import com.google.auto.service.AutoService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@AutoService(PantheonPlugin.class)
+public class RocksDBPlugin implements PantheonPlugin {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String NAME = "rocksdb";
+
+  private final RocksDBCLIOptions options;
+  private PantheonContext context;
+  private RocksDBKeyValueStorageFactory factory;
+  private RocksDBKeyValuePrivacyStorageFactory privacyFactory;
+
+  public RocksDBPlugin() {
+    this.options = RocksDBCLIOptions.create();
+  }
+
+  @Override
+  public void register(final PantheonContext context) {
+    LOG.info("Registering plugin");
+    this.context = context;
+
+    context
+        .getService(PicoCLIOptions.class)
+        .ifPresent(picoCLIOptions -> picoCLIOptions.addPicoCLIOptions(NAME, options));
+    LOG.info("Plugin registered.");
+  }
+
+  @Override
+  public void start() {
+    LOG.info("Starting plugin.");
+    LOG.debug("Applied configuration: {}", options.toString());
+    context
+        .getService(StorageService.class)
+        .ifPresentOrElse(
+            this::createAndRegister,
+            () -> LOG.error("Failed to start due to missing StorageService."));
+  }
+
+  @Override
+  public void stop() {
+    try {
+      if (factory != null) {
+        factory.close();
+      }
+
+      if (privacyFactory != null) {
+        privacyFactory.close();
+      }
+    } catch (final IOException e) {
+      LOG.error("Failed to stop plugin: " + e.getMessage(), e);
```

Comment:
nitpick: `LOG.error("....: {}". e.getMessage(), e);` can be a bit faster if it is available in log4j?

Reply:
Isn't that's only faster when the logging level is disabled (as it avoids the string concat)?

Reply:
done